test/client.js

```diff
@@ -0,0 +1,171 @@
+'use strict';
+
+const accounts = require('../index.js');
+
+const chai = require('chai');
+const chai_as_promised = require('chai-as-promised');
+const nock = require('nock');
+
+
+chai.should();
+chai.use(chai_as_promised);
+
+const configuration = {
+    base_url: 'https://localhost',
+    path: '/api',
+    client_id: 'test_id',
+    client_secret: 'test_secret'
+};
+
+const scopes = ['https://localhost/scope'];
+
+const payloads = {
+    client_credentials: {
+        client_id: configuration.client_id,
+        client_secret: configuration.client_secret,
+        scopes: scopes.join(' '),
+        grant_type: 'client_credentials'
+    },
+    refresh_token: {
+        client_id: configuration.client_id,
+        client_secret: configuration.client_secret,
+        grant_type: 'refresh_token',
+        refresh_token: 'jrrrefreshtoken'
+    }
+};
+
+const nocks = {
+    token: payload => {
+        return nock(configuration.base_url)
+            .post(configuration.path + '/oauth/token', payloads[payload]);
+    }
+};
+
+describe('generate_token', () => {
+
+    accounts.configure(configuration);
+
+    it('should get access token with specified scope', () => {
+        let accounts_nock = nocks.token('client_credentials')
+            .reply(200, {access_token: 'jrrtoken'});
+
+        accounts.configure({client_expiry: 0});
+
+        return accounts.generate_token(scopes)
+            .then(result => {
+                result.should.have.property('access_token');
+                result.access_token.should.be.a('string');
+                result.access_token.should.equal('jrrtoken');
+                accounts_nock.isDone().should.equal(true);
+            });
+    });
+
+    it('should get expired access token again with specified scope', () => {
+        let accounts_nock = nocks.token('client_credentials')
+            .reply(200, {access_token: 'jrrtoken'});
+
+        accounts.configure({client_expiry: 300});
+
+        return accounts.generate_token(scopes)
+            .then(result => {
+                result.should.have.property('access_token');
+                result.access_token.should.be.a('string');
+                result.access_token.should.equal('jrrtoken');
+                accounts_nock.isDone().should.equal(true);
+            });
+    });
+
+    it('should get cached access token with specified scope', () => {
```

Comment:
does this test depend on the previous tests?
I'm thinking that the test for this should be done by calling generate_token twice and getting the same result when the access_token is not yet expired. Then, have a test for getting a new token when the previous token expired.
Also, do not use arrow functions in mocha test cases https://mochajs.org/#arrow-functions
```js
it('should get the cached access token with specified scope', function () {
    ...
    accounts.generate_token(scopes)
        .then(assert_result)
        .then(accounts.generate_token)
        .then(assert_result)
});

it('should get a new access token with the specified scope when the previous token expires', function () {
    ...
    accounts.generate_token(scopes)
        .then(assert_result)
        .then(wait_for_expiry)
        .then(accounts.generate_token)
        .then(assert_result)
});
```

Reply:
Done:
0a41fea
b12a0ec